apps/server/src/templates/dto/template.dto.ts

```diff
@@ -0,0 +1,25 @@
+import { IsString, IsOptional, IsNotEmpty } from 'class-validator';
+
+export class IdParamDto {
+  @IsString()
+  @IsNotEmpty()
+  id: string;
+}
+
+export class TemplateDto {
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+
+  @IsString()
+  @IsNotEmpty()
+  subject: string;
+
+  @IsString()
+  @IsNotEmpty()
+  html: string;
+
+  @IsString()
+  @IsOptional()
+  apiKeyId?: string;
```

Comment:
Maybe we can avoid using the api key id here? Is this for private templates?

Reply:
The idea is to also have private templates. Instead of `apiKeyId` we can use `userId`, what do you think?

Reply:
yup I think `userId` makes more sense